build_tools/requirements-dev.txt

```diff
@@ -4,3 +4,4 @@ hatch-build-scripts
 hatch-requirements-txt
 hatch-sphinx
 hatch-vcs
+pre-commit
```

Comment:
`ruff` is not a dependency of `pre-commit`, so it is also needed here.

IMO, this requirements file is the right place for this.

Reply:
`ruff` is a dependency of the astral maintained hook used here, see https://github.com/astral-sh/ruff-pre-commit/blob/main/pyproject.toml.

I have set up a fresh virtual env and confirmed that this hook can be run without needing to install ruff explicitly, it's taken care of when running the hook for the first time.

Reply:
I was finally able to get this to trigger and it installed ruff automatically, so I agree, not necessary.